### PR TITLE
Set notification team type as a prefix to avoid long name duplicates

### PR DESCRIPTION
**Pull request by benbp**
We are having some issues where pipeline names that are just under the 64 character team name length limit, like `python - azure-communication-networktraversal - tests-weekly`. In these cases, they won't throw an error condition, but with the prefix separator `--` they hit the max and the team names are identical, when they need to be suffixed for either Notifications or Sync Notifications. If we move the suffix to a short prefix, it will avoid this name overlap issue altogether. Also, since the suffix often gets truncated down to a few characters anyway, I don't think we're losing any clarity of purpose, since the automation will also provide a clear description field like `purpose: ParentNotificationTeam` which gets shown in the team list view.

Trying to figure out the best way to test this, but I wanted to throw the PR up to get feedback on the approach.

### CODE REVIEW

File: tools/notification-configuration/notification-creator/NotificationConfigurator.cs
Comment (weshaggard) [inline tools/notification-configuration/notification-creator/NotificationConfigurator.cs:43]:
```suggestion
                    var parentTeam = await EnsureTeamExists(pipeline, "[notify]", TeamPurpose.ParentNotificationTeam, teams, persistChanges);
```

File: tools/notification-configuration/notification-creator/NotificationConfigurator.cs
Comment (weshaggard) [inline tools/notification-configuration/notification-creator/NotificationConfigurator.cs:44]:
```suggestion
                    var childTeam = await EnsureTeamExists(pipeline, "[owners]", TeamPurpose.SynchronizedNotificationTeam, teams, persistChanges);
```

**Review by weshaggard — APPROVED** [review]
I made a name suggestion it is 2 chars longer but I like them a little better.

**Comment by weshaggard** [comment]
@benbp once you change these names will the existing teams get updated correctly or we end up with a duplicate set of teams and need to clean-up the older ones?

**Comment by benbp** [comment]
> @benbp Ben Broderick Phillips FTE once you change these names will the existing teams get updated correctly or we end up with a duplicate set of teams and need to clean-up the older ones?

From reading the code, I think all the existing notification team names will be updated as it looks things up by team ID from the pipeline. I have to do some testing to verify though. I'd prefer not to have to change all the names, and the downside of having a prefix is that the sync and parent notification pair for a given pipeline doesn't get sorted together anymore. However I think keeping all the names consistent as opposed to a separate convention for >64 char names is preferable.

File: tools/notification-configuration/notification-creator/NotificationConfigurator.cs
Comment (benbp) [reply tools/notification-configuration/notification-creator/NotificationConfigurator.cs:44]: This prefix solution still kicks the can down the road though, the moment we have a pipeline name with a few more characters we're going to hit this problem yet again. The only other solutions I can think of is to not rely on preserving the pipeline name or having some shorthand for the language/test type/separator.

Comment (weshaggard) [reply tools/notification-configuration/notification-creator/NotificationConfigurator.cs:44]: After chatting I think we landed on changing this to something like:
1. notify team should match the name of the pipeline. If the name is too long then truncate and add pipeline id at the end to make sure we don't have conflicts.
2. sync team we should just use the pipeline id and use something like "codeowners sync for <id>"

Comment (benbp) [reply tools/notification-configuration/notification-creator/NotificationConfigurator.cs]: Implementing ^^ but I'm putting the pipeline ID at the start of the name, that way the parent and sync notification teams still sort next to each other in the list view for ease of use.

Parent name format: `[<pipeline id>] <pipeline name>]`
Sync name format: `[<pipeline id>] Code owners sync notifications`

Dry run log output example for the new change (handling renaming of existing pipelines):

```
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Found team with matching pipeline id 1034 but different name 'python - sql -- Notifications', expected '[1034] python - sql'. Purpose = 'ParentNotificationTeam'
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Update Team for Pipeline PipelineId = 1034 Purpose = ParentNotificationTeam Name = '[1034] python - sql'
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Found team with matching pipeline id 1034 but different name 'python - sql -- Sync Notifications', expected '[1034] Code owners sync notifications'. Purpose = 'SynchronizedNotificationTeam'
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Update Team for Pipeline PipelineId = 1034 Purpose = SynchronizedNotificationTeam Name = '[1034] Code owners sync notifications'
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      GetDescriptor Id = fb8f4d34-1636-439c-a957-c14881afa7bf
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      GetDescriptor Id = e64231d0-8547-4e65-b61c-bf91456affda
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]                                                                                                                                                                                                                                                                                                                                                               => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      CheckMembership ParentId = vssgp.Uy0xLTktMTU1MTM3NDI0NS03MjEyMjY4NDEtNDc1NTgxMjYxLTI3MDQxNzIzOTItMzQ2Mzk5NTc2NS0xLTg2OTkyODc4Mi0zMDgwNjE3Mjg4LTI2NzI1MjM3NjItMTgxMDM3MDM5NQ ChildId = vssgp.Uy0xLTktMTU1MTM3NDI0NS03MjEyMjY4NDEtNDc1NTgxMjYxLTI3MDQxNzIzOTItMzQ2Mzk5NTc2NS0xLTM4Mjk3Njg0NjItMzk4NTU3MDExNS0yOTI5NzEwMDIyLTIyOTY1ODc1NzE
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Child In Parent ParentId = fb8f4d34-1636-439c-a957-c14881afa7bf, ChildId = e64231d0-8547-4e65-b61c-bf91456affda, IsInTeam = True
info: Azure.Sdk.Tools.NotificationConfiguration.Services.AzureDevOpsService[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      GetSubscriptionsAsync TargetId = fb8f4d34-1636-439c-a957-c14881afa7bf
info: Azure.Sdk.Tools.NotificationConfiguration.NotificationConfigurator[0]
      => Evaluate Pipeline Name = python - sql, Path = \python, Id = 1034
      Team Is Subscribed TeamName = [1034] python - sql PipelineId = 1034
```

**Review by weshaggard — APPROVED** [review]

**Pull request merged**
